plugin/v2/healthz.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v2
+
+import (
+	"context"
+	"fmt"
+
+	"net"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/google/uuid"
+	kmspb "google.golang.org/api/cloudkms/v1"
+	"google.golang.org/grpc"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"github.com/GoogleCloudPlatform/k8s-cloudkms-plugin/plugin"
+)
+
+type HealthZ struct {
+	plugin.HealthZ
+}
+
+func NewHealthChecker(keyName string, keyService *kmspb.ProjectsLocationsKeyRingsCryptoKeysService,
+	unixSocketPath string, callTimeout time.Duration, servingURL *url.URL) *HealthZ {
+
+	return &HealthZ{
+		plugin.HealthZ{
+			KeyName:        keyName,
+			KeyService:     keyService,
+			UnixSocketPath: unixSocketPath,
+			CallTimeout:    callTimeout,
+			ServingURL:     servingURL,
+		},
+	}
+
+}
+
+// Serve creates http server for hosting healthz.
+func (h *HealthZ) Serve() chan error {
```

Comment:
I'm not seeing a major difference between this and the v1 health check. There seems to be a lot of duplication.